utils/hct/hctbuild.cmd

```diff
@@ -184,13 +179,19 @@ if "%1"=="-dxc-cmake-ends-include" (
   shift /1
 )
 
-rem If only VS 2017 is available, pick that by default.
-if "%BUILD_VS_VER%"=="2015" (
-  reg query HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\DevDiv\vs\Servicing\14.0\devenv /v Install /reg:32 1>nul 2>nul
-  if errorlevel 1 (
-    echo Visual Studio 2015 not available, setting up build for Visual Studio 2017.
-    set BUILD_GENERATOR=Visual Studio 15 2017
-    set BUILD_VS_VER=2017
+rem If only VS 2019 is available, pick that by default.
+if "%BUILD_VS_VER%"=="2017" (
+  if not exist "%ProgramFiles(x86)%\Microsoft Visual Studio\2017" (
+    echo Visual Studio 2017 not available, setting up build for Visual Studio 2019.
+    set BUILD_GENERATOR=Visual Studio 16 2019
+    set BUILD_VS_VER=2019
+  )
+)
+
+if "%BUILD_VS_VER%"=="2019" (
+  if not exist "%ProgramFiles(x86)%\Microsoft Visual Studio\2019" (
+    echo Visual Studio 2019 not available. Use hctbuild -vs2017 to build with Visual Studio 2017.
```

Comment:
In this case and the 2017 case, I think it would be useful if the error message included the path it failed to find which led to this conclusion. If someone installed VS somewhere bizarre, they would at least immediately know the cause. 

I remember someone having that problem with an unconventional VS installation location.

Reply:
This too.